### PR TITLE
Add marketplaces to tests in Tests/conf.json

### PR DESCRIPTION
**Pull request by BEAdi**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-8938)
related: https://github.com/demisto/content/pull/29835, https://github.com/demisto/demisto-sdk/pull/3875

## Description
1. Add support for running tests on a specific marketplace.
2. Ignored integrations from `Tests/conf.json` will be ignored in the test collector.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Tests/scripts/collect_tests/collect_tests.py [hunk 1]
```diff
@@ -239,13 +239,21 @@ def _validate_collection(
...
+                # test_playbook.implementing_integrations is from id set (always empty), test_integrations is from conf.json
+                for integration in test_playbook.implementing_integrations + test_integrations:
                     if reason := conf.skipped_integrations.get(integration):  # type: ignore[union-attr, assignment]
                         raise SkippedTestException(
                             test_name=test,
                             skip_place='conf.json (integrations)',
                             skip_reason=f'{test=} uses {integration=}, which is skipped ({reason=})'
                         )
+                test_marketplaces = conf.tests_to_marketplace_set[test]  # type: ignore[union-attr]
+                logger.info(f'HERE MARKETPLACES: {test_marketplaces=}, {conf.marketplace=}')  # type: ignore[union-attr]
```
Comment (dorschw) [inline Tests/scripts/collect_tests/collect_tests.py:252]:
```suggestion
                logger.info(f'{test_marketplaces=}, {conf.marketplace=}')  # type: ignore[union-attr]
```

File: Tests/scripts/collect_tests/collect_tests.py [hunk 2]
```diff
@@ -239,13 +239,21 @@ def _validate_collection(
...
+                if test_marketplaces and conf.marketplace not in test_marketplaces:  # type: ignore[union-attr]
```
Comment (dorschw) [inline Tests/scripts/collect_tests/collect_tests.py:253]:
```suggestion
                if test_marketplaces and (conf.marketplace not in test_marketplaces):  # type: ignore[union-attr]
```
just for readability

File: Tests/scripts/collect_tests/collect_tests.py [hunk 3]
```diff
@@ -1470,7 +1478,7 @@ def _collect(self) -> CollectionResult | None:
 
     marketplace = MarketplaceVersions(args.marketplace)
 
-    nightly = args.nightly
+    nightly = True
```
Comment (dorschw) [inline Tests/scripts/collect_tests/collect_tests.py:1481]: for testing I assume?

File: Tests/scripts/collect_tests/exceptions.py [hunk 4]
```diff
@@ -121,3 +121,9 @@ def __init__(self, test_name: str, skip_place: str, skip_reason: Optional[str] =
 class PrivateTestException(InvalidTestException):
     def __init__(self, test_name: str):
         super().__init__(test_name, 'test is private')
+
+
+class IncompatibleTestMarketplaceException(InvalidTestException):
+    def __init__(self, test_name: str, test_marketplaces: set[MarketplaceVersions], expected_marketplace: MarketplaceVersions):
+        super().__init__(test_name, f'test marketplace values are {", ".join(test_marketplaces)}, '
+                                    f'which is not compatible with expected marketplace {expected_marketplace}')
```
Comment (dorschw) [inline Tests/scripts/collect_tests/exceptions.py:129]: super minor
```suggestion
                                    f', incompatible with {expected_marketplace=}')
```

**Review by dorschw — COMMENTED** [review]
🚀

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**